zotify: init at 0.6.13

bwkam opened the pull request:
## Description of changes

Closes #310331 

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

* added zotify
* added pwinput 
* added a new maintainer entry

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

bwkam committed:
python311Packages.pwinput: init at 1.3.0

bwkam committed:
maintainers: add bwkam

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3918

jtbx commented on maintainers/maintainer-list.nix:
```diff
@@ -3054,6 +3054,12 @@
+  bwkam = {
+    email = "beshoykamel391@gmail.com";
+    github = "bwkam";
+    githubId = 	91009118;
```
```suggestion
    githubId = 91009118;
```

jtbx commented on pkgs/development/python-modules/pwinput/default.nix:
```diff
@@ -0,0 +1,29 @@
+  doCheck = false;
+
+  meta = {
+    description = "A cross-platform Python module that displays **** for password input. Works on Windows, unlike getpass. Formerly called stdiomask.";
```
A lot of this information is irrelevant especially since Nix does not run on Windows
```suggestion
    description = "Python module that masks password input";
```

jtbx commented on pkgs/tools/misc/zotify/default.nix:
```diff
@@ -0,0 +1,59 @@
+
+buildPythonPackage rec {
+  pname = "zotify";
+  version = "0.16.13";
```
```suggestion
  version = "0.16.13";

```

jtbx commented on pkgs/top-level/python-packages.nix:
```diff
@@ -9727,6 +9727,7 @@ self: super: with self; {
 
   pandas-stubs = callPackage ../development/python-modules/pandas-stubs { };
 
+
```
```suggestion
```

jtbx replied:
Please remove this newline added by this specific commit

jtbx commented on pkgs/top-level/python-packages.nix:
```diff
@@ -17407,6 +17410,8 @@ self: super: with self; {
     inherit (pkgs) zopfli;
   };
 
+  zotify = callPackage ../tools/misc/zotify { };
+
```
Why is zotify here?

bwkam replied:
I wasn't sure of the best place to put it, so I looked for where youtube-dl is and had it there. Where does it have to be instead?

jtbx replied:
It should usually be in pkgs/top-level/all-packages.nix. I think python libraries are meant to go here.

jtbx reviewed (changes requested):
Hey thanks for making your first contribution! I have some notes but otherwise it looks good. :smile:

bwkam committed:
zotify: init at 0.6.13

jtbx commented on pkgs/by-name/zo/zotify/package.nix:
```diff
@@ -0,0 +1,52 @@
+  doCheck = false;
+
+  meta = {
+    description = "A fast and customizable music and podcast downloader.";
```
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes
```suggestion
    description = "Fast and customizable music and podcast downloader";
```

jtbx commented on pkgs/top-level/all-packages.nix:
```diff
@@ -7738,6 +7738,10 @@ with pkgs;
 
   zoxide = callPackage ../tools/misc/zoxide { };
 
+  zotify = callPackage ../by-name/zo/zotify/package.nix {
+    inherit fetchFromGitHub python3Packages protobuf_21 lib;
+  };
+
```
callPackage will do all of this for you
```suggestion
  zotify = callPackage ../by-name/zo/zotify/package.nix { };

```

bwkam replied:
CI fails when I do that

jtbx replied:
Oh that's right, when using the pkgs/by-name hierarchy you can remove any entry from all-packages.nix altogether unless it calls callPackage with extra attributes.

bwkam replied:
Ah, so do I add the attributes back?

jtbx reviewed (approved):
Neat thanks a lot! Looks good <3

itslychee commented on pkgs/by-name/zo/zotify/package.nix:
```diff
@@ -0,0 +1,52 @@
+    substituteInPlace ./setup.cfg --replace-fail "protobuf==3.20.1" "protobuf"
+  '';
+
+  pyproject = true;
```
Move this around `version`, just for readability purposes!

bwkam replied:
I'm not sure if I understood that correctly, can you check if's good now?

itslychee commented on pkgs/by-name/zo/zotify/package.nix:
```diff
@@ -0,0 +1,52 @@
+
+  pyproject = true;
+
+  nativeBuildInputs = [ python3Packages.setuptools ];
```
This should be `build-systems`, there have been some relatively recent changes surrounding the python wrapper 

Read around here
https://nixos.org/manual/nixpkgs/unstable/#buildpythonpackage-parameters

itslychee commented on pkgs/by-name/zo/zotify/package.nix:
```diff
@@ -0,0 +1,52 @@
+
+  nativeBuildInputs = [ python3Packages.setuptools ];
+
+  propagatedBuildInputs =
```
Same as the `nativeBuildInputs` comment, this should be `dependencies`, please read this link.

https://nixos.org/manual/nixpkgs/unstable/#buildpythonpackage-parameters

itslychee commented on pkgs/development/python-modules/pwinput/default.nix:
```diff
@@ -0,0 +1,29 @@
+    hash = "sha256-yhqL0G4ohy11Hb1BMthjcSfCW0COo6NJN3MUpUkUJvM=";
+  };
+
+  nativeBuildInputs = [ setuptools ];
+  propagatedBuildInputs = [ pytest ];
```
https://nixos.org/manual/nixpkgs/unstable/#buildpythonpackage-parameters

Also, shouldn't pytest be in `nativeCheckInputs`, but what's the purpose for it being here if `doCheck` is set to `false`?

bwkam replied:
Yeah, I forgot about thatl.

itslychee reviewed (changes requested):
congratulations on your first PR to nixpkgs, i've added a few notes!

itslychee reviewed (approved):
looks good!

Aleksanaa commented on pkgs/development/python-modules/pwinput/default.nix:
```diff
@@ -0,0 +1,27 @@
+
+  nativeBuildInputs = [ setuptools ];
+
+  doCheck = false;
```
Please provide a reason in the comment.

bwkam replied:
```
 # the tests fail, but the library works pretty well; has been tested with zotify
 ```
is that food?

Aleksanaa replied:
That's too long, just "Requires graphical environment to use pyautogui"

bwkam replied:
Done

bwkam replied:
Done.

drupol commented on pkgs/by-name/zo/zotify/package.nix:
```diff
@@ -0,0 +1,52 @@
+    hash = "sha256-KA+Q4sk+riaFTybRQ3aO5lgPg4ECZE6G+By+x2uP/VM=";
+  };
+
+  patchPhase = ''
```
I think you should use `postPatch`.

drupol commented on pkgs/by-name/zo/zotify/package.nix:
```diff
@@ -0,0 +1,52 @@
+
+  doCheck = false;
+
+  meta = {
```
No `meta.mainProgram` ?

drupol commented:
Last thing, did you use `nixfmt-rfc-style` to format the new files you're introducing?

bwkam commented:
> Last thing, did you use `nixfmt-rfc-style` to format the new files you're introducing?

I think so.

drupol commented on pkgs/by-name/zo/zotify/package.nix:
```diff
@@ -0,0 +1,52 @@
+  src = fetchFromGitHub {
+    owner = "zotify-dev";
+    repo = "zotify";
+    rev = "5da27d32a1f522e80a3129c61f939b1934a0824a";
```
Why not use the version tag here?

bwkam replied:
I didn't find any version tags on their repo.

drupol replied:
Fair enough!

Qyriad replied:
Putting a comment explaining that then would probably be good, even if it's just "Repository has no version tags."

bwkam replied:
Done :+1:

drupol commented on pkgs/by-name/zo/zotify/package.nix:
```diff
@@ -0,0 +1,52 @@
+  };
+
+  postPatch = ''
+    substituteInPlace ./setup.cfg --replace-fail "protobuf==3.20.1" "protobuf"
```
Can't we relax dependencies here instead of doing a substitute?

Tom-Hubrecht replied:
You need to add `python3Packages.pythonRelaxDepsHook` to `nativeBuildInputs` and set `pythonRelaxDeps = [ "protobuf" ];`

bwkam replied:
It's working now, I didn't know you had to add it to `nativeBuildInputs`, thanks! I'll modify it now.

bwkam replied:
Yeah it still fails.

Tom-Hubrecht replied:
If you use https://bin.hubrecht.ovh/?48418a0288c4d03d#5sNySCHkptNWGCnRpfxupgjKq3JdE6Xb1G5w8ovWrQxc it seems to build fine

bwkam replied:
Oh, yeah that builds fine for me too. Thanks, I pushed the new modifications.

GaetanLepage commented on pkgs/development/python-modules/pwinput/default.nix:
```diff
@@ -0,0 +1,28 @@
+    hash = "sha256-yhqL0G4ohy11Hb1BMthjcSfCW0COo6NJN3MUpUkUJvM=";
+  };
+
+  nativeBuildInputs = [ setuptools ];
```
```suggestion
  build-system = [ setuptools ];
```

drupol commented on pkgs/development/python-modules/pwinput/default.nix:
```diff
@@ -0,0 +1,28 @@
+  version = "1.0.3";
+
+  src = fetchPypi {
+    inherit pname version;
```
```suggestion
    inherit version;
    pname = "pwinput";
```

drupol replied:
I don't think we should reuse `pname` for anything else.

bwkam replied:
Alright

natsukium commented on pkgs/by-name/zo/zotify/package.nix:
```diff
@@ -0,0 +1,46 @@
+    protobuf
+  ];
+
+  doCheck = false;
```
Why do you disable tests here?

Also, please add `pythonImportsCheck` for a simple check.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck-using-pythonimportscheck

bwkam replied:
There were no tests, but I guess I shouldn't have explicitly disabled it. I added `pythonImportsCheck` now.

natsukium commented on pkgs/development/python-modules/pwinput/default.nix:
```diff
@@ -0,0 +1,28 @@
+buildPythonPackage rec {
+  pname = "pwinput";
+  version = "1.0.3";
+
```
Could you add `pyproject = true` to avoid a legacy `setup.py` build?

> Note that the pyproject format falls back to using setuptools, so you can use pyproject = true even if the package only has a setup.py.

> When unset, the legacy setuptools hooks are used for backwards compatibility.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

natsukium commented on pkgs/development/python-modules/pwinput/default.nix:
```diff
@@ -0,0 +1,28 @@
+
+  # Requires graphical environment to use pyautogui
+  doCheck = false;
+
```
Please add `pythonImportsCheck`.

bwkam replied:
`pwinput` doesn't use any third-party `modules`, but I've added the native modules it imports.

natsukium commented on pkgs/development/python-modules/pwinput/default.nix:
Could you update the commit message from `pwinput: init at 1.3.0` to `python311Packages.pwinput: init at 1.3.0`?

> Commit names of Python libraries should reflect that they are Python libraries, so write for example python311Packages.numpy: 1.11 -> 1.12. It is highly recommended to specify the current default version to enable automatic build by ofborg.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines-contributing-guidelines

natsukium commented on pkgs/by-name/zo/zotify/package.nix:
```diff
@@ -0,0 +1,46 @@
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "zotify";
+  version = "0.16.13";
```
```suggestion
  version = "0.6.13";
```
https://github.com/zotify-dev/zotify/blob/5da27d32a1f522e80a3129c61f939b1934a0824a/setup.cfg#L3

bwkam replied:
Sorry for the typo. Fixed it.

fabaff commented on pkgs/by-name/zo/zotify/package.nix:
```diff
@@ -0,0 +1,58 @@
+{
+  fetchFromGitHub,
+  python3Packages,
+  lib,
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "zotify";
```
```suggestion
{
  fetchFromGitHub,
  python3,
  lib,
}:

python3.pkgs.buildPythonApplication rec {
  pname = "zotify";
```

Will make the diff in the future significantly smaller if an override is needed.

bwkam replied:
I don't quite understand this comment, can you please explain how that's going to make the diff smaller?

fabaff replied:
> ...how that's going to make the diff smaller?

When an override is needed. `protobuf` is already relaxed, thus there is a chance that we need an override for this sooner than later.

bwkam replied:
why would we need an override though?

fabaff commented on pkgs/by-name/zo/zotify/package.nix:
```diff
@@ -0,0 +1,58 @@
+  src = fetchFromGitHub {
+    owner = "zotify-dev";
+    repo = "zotify";
+    # repository has no version tags
```
```suggestion
    # repository has no version tags
    # https://github.com/zotify-dev/zotify/issues/124
```

fabaff commented on pkgs/by-name/zo/zotify/package.nix:
```diff
@@ -0,0 +1,58 @@
+    hash = "sha256-KA+Q4sk+riaFTybRQ3aO5lgPg4ECZE6G+By+x2uP/VM=";
+  };
+
+  build-system = [ python3Packages.setuptools ];
```
```suggestion
  build-system = [ python3.pkgs.setuptools ];
```

or `build-system = with python3.pkgs; [ setuptools ];`...more of a personal preference.

SomeoneSerge replied:
Unless `python3.pkgs` has been recently made to work with splicing, it isn't. The manual was specifically updated to advertise `python3Packages` instead of `python3.pkgs`: https://nixos.org/manual/nixpkgs/unstable/#buildpythonapplication-function

SomeoneSerge replied:
Sorry I wasn't more precise: https://github.com/NixOS/nixpkgs/commit/70aa345bec1135ea9524cd893aea1f6e3dd82387

Qyriad replied:
Link to the actual issue: https://github.com/NixOS/nixpkgs/issues/211340.

This still is not really documented, though

bwkam replied:
I'm not sure which to use now. Most python packages I saw use `python3Packages`, so that's what I went with.

SomeoneSerge replied:
I'm pretty confident that in the current situation we should keep `python3Packages`

drupol replied:
@fabaff ? @mweinelt ? @natsukium ? What's your POV on this?

mweinelt replied:
python3Packages, for splicing reasons, as mentioned by SomeoneSerge.

AndersonTorres replied:
For reference about this dreadful splicing issue, yay!, the issues are

#211340 (python but general)
#204303 (top-level)

fabaff commented on pkgs/by-name/zo/zotify/package.nix:
```diff
@@ -0,0 +1,58 @@
+
+  build-system = [ python3Packages.setuptools ];
+
+  nativeBuildInputs = [ python3Packages.pythonRelaxDepsHook ];
```
```suggestion
  nativeBuildInputs = [ python3.pkgs.pythonRelaxDepsHook ];
```

fabaff commented on pkgs/by-name/zo/zotify/package.nix:
```diff
@@ -0,0 +1,58 @@
+
+  pythonRelaxDeps = [ "protobuf" ];
+
+  dependencies = with python3Packages; [
```
```suggestion
  dependencies = with python3.pkgs; [
```

fabaff commented on pkgs/by-name/zo/zotify/package.nix:
```diff
@@ -0,0 +1,58 @@
+    protobuf
+  ];
+
+  pythonImportsCheck = [
+    "zotify.album"
+    "zotify.app"
+    "zotify.config"
+    "zotify.const"
+    "zotify.loader"
+    "zotify.playlist"
+    "zotify.podcast"
+    "zotify.termoutput"
+    "zotify.track"
+    "zotify.utils"
+    "zotify.zotify"
+  ];
```
If the import of `zotify` fails then `zotify.album`, `zotify.app`, etc. will fail too. 

```suggestion
  pythonImportsCheck = [
    "zotify"
  ];
```

Checking the import of `zotify.album`, `zotify.app`, etc. would make sense if they depend on optional dependencies or have other requirements.

fabaff commented on pkgs/development/python-modules/pwinput/default.nix:
```diff
@@ -0,0 +1,32 @@
+  pname = "pwinput";
+  version = "1.0.3";
+
+  pyproject = true;
+
```
The chance is pretty low that somebody will test the module on Python releases which are EOL.

```suggestion
  pyproject = true;

  disabled = pythonOlder "3.7";
```

cottand committed:
nomad, nomad_1_7, nomad_1_8: build with buildGo122Module

paparodeo commented:
this looks like it should get rebased to master and the files under `python-modules` need to be run under nixfmt otherwise i believe the python format github check will break for following PRs

drupol commented:
Setting this PR to draft until everything is fixed, feel free to mark it as ready when you believe it's ready.

leon-erd committed:
onlyoffice: drop 7.2.0 version

NukaDuka committed:
prometheus-pve-exporter: 3.2.4 -> 3.4.3

From version 3.2.5, the package's name was changed upstream to use underscores (_) instead of dashes (-). Even though `pname` was changed to reflect this, the nix package itself will still be referenced by `prometheus-pve-exporter`, thus maintaining compatibility.

NukaDuka committed:
nixos/prometheus-pve-exporter: added new options introduced in v3.4.3

r-ryantm committed:
rocksdb: 9.3.1 -> 9.4.0

Enzime committed:
regreet: add options for controlling theming

NukaDuka committed:
grafanaPlugins.grafana-oncall-app: 1.7.1 -> 1.8.5

reckenrode committed:
nodePackages.node2nix: use top-level cctools in generated files

jpds committed:
apacheKafka: 3.7.0 → 3.7.1

cottand committed:
nomad_1_8: 1.8.0 -> 1.8.2

cottand committed:
nomad_1_5: deprecate

veprbl committed:
python312Packages.botorch: relax gpytorch version requirement

and migrate to PEP517 lingo for dependencies.

Co-Authored-By: Martin Weinelt <hexa@darmstadt.ccc.de>

veprbl committed:
python312Packages.ax: relax botorch version requirement

r-ryantm committed:
fetchmail: 6.4.38 -> 6.4.39

r-ryantm committed:
libgbinder: 1.1.39 -> 1.1.40

eclairevoyant committed:
yt-dlp: move to by-name

eclairevoyant committed:
yt-dlp: add curl impersonation support

reckenrode committed:
gcc12: update aarch64-darwin patch for GCC 12.4

reckenrode committed:
gcc10: enable build on aarch64-darwin

r-ryantm committed:
cargo-tarpaulin: 0.30.0 -> 0.31.0

r-ryantm committed:
docker-buildx: 0.16.0 -> 0.16.1

GaetanLepage committed:
python311Packages.lightning-utilities: 0.11.5 -> 0.11.6

Diff: https://github.com/Lightning-AI/utilities/compare/refs/tags/v0.11.5...v0.11.6

Changelog: https://github.com/Lightning-AI/utilities/releases/tag/v0.11.6

r-ryantm committed:
kcov: 42 -> 43

r-ryantm committed:
wasmer: 4.3.4 -> 4.3.5

guoard committed:
docker-compose: 2.29.0 -> 2.29.1

ju1m committed:
nixos/radicle: fix support of absolute file paths for SSH keys

r-ryantm committed:
openrefine: 3.8.1 -> 3.8.2

malob committed:
python311Packages.pandas-stubs: 2.2.0.240218 -> 2.2.2.240603, update/fix deps and tests

anthonyroussel committed:
python311Packages.python-barbicanclient: init at 7.0.0

https://wiki.openstack.org/wiki/Barbican
https://opendev.org/openstack/python-barbicanclient

anthonyroussel committed:
python311Packages.openstacksdk: 3.1.0 -> 3.3.0

https://github.com/openstack/openstacksdk/compare/3.1.0...3.3.0

YvesStraten committed:
skimpdf: 1.7.2 -> 1.7.3

NickCao committed:
libkrunfw: 4.0.0-unstable-2024-06-10 -> 4.2.0

Diff: https://github.com/containers/libkrunfw/compare/12236fa4caa42423ff3081b6179aa0a5f37c67c9...v4.2.0

NickCao committed:
libkrun: 1.9.3 -> 1.9.4

Diff: https://github.com/containers/libkrun/compare/refs/tags/v1.9.3...v1.9.4

r-ryantm committed:
libcdada: 0.5.2 -> 0.6.0

r-ryantm committed:
pinentry-bemenu: 0.13.2 -> 0.14.0

r-ryantm committed:
cozette: 1.24.1 -> 1.25.0

r-ryantm committed:
process-compose: 1.6.1 -> 1.9.0

r-ryantm committed:
cargo-deny: 0.14.24 -> 0.15.0

r-ryantm committed:
cargo-workspaces: 0.3.2 -> 0.3.5

r-ryantm committed:
prometheus-fastly-exporter: 8.1.0 -> 9.0.0

r-ryantm committed:
havn: 0.1.12 -> 0.1.13

ryand56 committed:
freetube: 0.21.1 -> 0.21.2

r-ryantm committed:
chafa: 1.14.1 -> 1.14.2

r-ryantm committed:
release-plz: 0.3.74 -> 0.3.79

r-ryantm committed:
cargo-mutants: 24.7.0 -> 24.7.1

r-ryantm committed:
ktop: 0.3.6 -> 0.3.7

r-ryantm committed:
kubectl-ktop: 0.3.6 -> 0.3.7

r-ryantm committed:
fedimint: 0.3.2 -> 0.3.3

r-ryantm committed:
openjph: 0.14.2 -> 0.15.0

r-ryantm committed:
far2l: 2.6.2 -> 2.6.3

TheDelus committed:
xournalpp: fix missing icons that prevent start

Commit:
pulumi-bin: 3.126.0 -> 3.127.0

r-ryantm committed:
bottom: 0.9.6 -> 0.9.7

r-ryantm committed:
cryptpad: 2024.6.0 -> 2024.6.1

r-ryantm committed:
downonspot: 0.5.0 -> 0.5.1

r-ryantm committed:
automatic-timezoned: 2.0.22 -> 2.0.25

r-ryantm committed:
faas-cli: 0.16.30 -> 0.16.31

r-ryantm committed:
ginkgo: 2.19.0 -> 2.19.1

r-ryantm committed:
imsprog: 1.4.2 -> 1.4.3

r-ryantm committed:
nuclei-templates: 9.9.1 -> 9.9.2

GaetanLepage committed:
python311Packages.transformers: 4.43.2 -> 4.43.3

Diff:
https://github.com/huggingface/transformers/compare/v4.43.2...v4.43.3

Changelog:
https://github.com/huggingface/transformers/releases/tag/v4.43.3

jukremer commented:
How can I push to this PR? I fixed the formatting and merged into master. 
As far as I can tell @bwkam already resolved everything else.

Should I create a new PR?

Mynacol committed:
libreoffice: Add meta.mainProgram

So users can leverage `lib.getExe`.
The expectation for the main program is the general libreoffice
program, not specific binaries for e.g. Writer or Impress.

r-ryantm committed:
qlog: 0.37.1 -> 0.37.2

r-ryantm committed:
woodpecker-plugin-git: 2.5.1 -> 2.5.2

pyrox0 committed:
python312Packages.pypass: Remove nose dependency

pyrox0 committed:
python312Packages.pypass: Modernize

bwkam commented:
> How can I push to this PR? I fixed the formatting and merged into master. As far as I can tell @bwkam already resolved everything else.
> 
> Should I create a new PR?

I've resolved everything, but this has been stale for a while, and I gave up on it getting merged. You can try with a new PR.

jukremer commented:
> I've resolved everything, but this has been stale for a while, and I gave up on it getting merged. You can try with a new PR.

pkgs/development/python-modules/pwinput/default.nix needs to be formatted.

jukremer commented on pkgs/development/python-modules/pwinput/default.nix:
```diff
@@ -0,0 +1,35 @@
+    hash = "sha256-yhqL0G4ohy11Hb1BMthjcSfCW0COo6NJN3MUpUkUJvM=";
+  };
+
+  build-system = [setuptools];
```
```suggestion
  build-system = [ setuptools ];
```

jukremer commented on pkgs/development/python-modules/pwinput/default.nix:
```diff
@@ -0,0 +1,35 @@
+
+  build-system = [setuptools];
+
+  pythonImportsCheck = ["pwinput"];
```
```suggestion
  pythonImportsCheck = [ "pwinput" ];
```

jukremer commented on pkgs/development/python-modules/pwinput/default.nix:
```diff
@@ -0,0 +1,35 @@
+    description = "Python module that masks password input";
+    homepage = "https://github.com/asweigart/pwinput";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [bwkam];
```
```suggestion
    maintainers = with lib.maintainers; [ bwkam ];
```

bwkam commented:
I'll fix in a bit if you want

jukremer reviewed (changes requested):
Fixed formatting

davidkna committed:
starship: 1.19.0 -> 1.20.0

r-ryantm committed:
symfony-cli: 5.9.1 -> 5.10.2

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/frustrations-about-splicing/49607/1

nicoonoclaste committed:
nixos/nix-channel: don't set `nix-path` (#327683)

Otherwise, the empty path in `nix.conf` takes precedence over `NIX_PATH`,
and by extension the `nix.nixPath` configuration option.

Introduced in 61afc4d1662fe426f02b28e386f2e053f887b6d6.

RafaelKr committed:
mailpit: implement `passthru.updateScript` (#324302)

D3vil0p3r committed:
caido: 0.33.0 -> 0.39.0 (#328499)

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

r-ryantm committed:
discord-development: 0.0.23 -> 0.0.24

bwkam commented:
I applied your formatting changes to `pkgs/development/python-modules/pwinput/default.nix`, but the CI still fails. I'm not sure why,

anthonyroussel committed:
python311Packages.python-designateclient: init at 6.0.1 (#328506)

jukremer commented:
> [I applied your formatting changes](https://github.com/NixOS/nixpkgs/pull/310701/commits/b0e0716d378980a2ac81438177f4c0be8d4ad625) to `pkgs/development/python-modules/pwinput/default.nix`, [but the CI still fails](https://github.com/NixOS/nixpkgs/actions/runs/10117044275/job/27981209604?pr=310701). I'm not sure why.

I'm not sure why, I formatted both files again locally but it doesn't make any changes.

bwkam commented:
Oh god. I'm so sorry everyone
